webapp/src/idbworkspace.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * A workspace implementation that uses IndexedDB directly (bypassing PouchDB), to support WKWebview where PouchDB
+ * doesn't work.
+ */
+
+type Header = pxt.workspace.Header;
+type ScriptText = pxt.workspace.ScriptText;
+type WorkspaceProvider = pxt.workspace.WorkspaceProvider;
+
+interface StoredText {
+    id: string;
+    files: pxt.Map<string>;
+    _rev: string;
+};
+
+const TEXTS_TABLE = "texts";
+const HEADERS_TABLE = "headers";
+const KEYPATH = "id";
+
+let _db: pxtc.Util.IDBWrapper;
+
+function getDbAsync(): Promise<pxtc.Util.IDBWrapper> {
+    if (_db) {
+        return Promise.resolve(_db);
+    }
+
+    _db = new pxtc.Util.IDBWrapper("__pxt_idb_workspace", 1, (ev, r) => {
```

Comment:
does "Reset" delete this table?

Reply:
Reset only clears the stores (texts and headers), the database and both stores will all still exist (but the stores will be empty)